test/cmd/crd.sh

```diff
@@ -292,7 +292,7 @@ run_non_native_resource_tests() {
   # Start watcher in background with process substitution,
   # so we can read from stdout asynchronously.
   kube::log::status "Testing CustomResource watching"
-  exec 3< <(kubectl "${kube_flags[@]}" get bars --request-timeout=1m --watch-only -o name & echo $! ; wait)
```

Comment:
Hmm.... this worries me, how is `request-timeout` handled with the new watches of yours? This is a significant API change.

Reply:
`request-timeout` should be orthogonal to your `timeout` they both should work just fine and you should be able to pass both.

Reply:
I'd not be worried. I don't think is api-change.

This test case is testing that `kubectl get` is broken, fixing it seems like desirable bug fix.

`--request-timeout` means *The length of time to wait before giving up on a single server request*. 

Watching is not a single request process. Ending a WATCH call should have never ended up watching for resource changes.  It will still timeout the single watch calls as it did before but now we can recover.

Reply:
Have you verified behavior before and with your fix?

Reply:
I did. Failed before because there was a bug, now it doesn't. 

(I am not saying the behaviour doesn't differ, jut that changing a behaviour from buggy->working isn't api change.)